### PR TITLE
tl.find verbose logs off by default for vstest task

### PR DESCRIPTION
**Pull request by ShreyasRmsft**
doing this for multiple reasons:

Log file sizes are bloating up like crazy
Too much clutter in the logs
Additional time spent logging all this
customers might be more willing to directly attach the logs zip since test sources are no longer explicitly listed in the logs

### CODE REVIEW

**Comment by ShreyasRmsft** [comment]
Contains other minor cleanup for the vstest task

File: Tasks/VsTestV2/nondistributedtest.ts [hunk 1]
```diff
@@ -116,30 +102,49 @@ export class NonDistributedTest {
...
+
     private createTestSourcesFile(): string {
         try {
             console.log(tl.loc('UserProvidedSourceFilter', this.sourceFilter.toString()));
-    
-            const sources = tl.findMatch(this.inputDataContract.TestSelectionSettings.SearchFolder, this.sourceFilter);
+
+            const sources = this.getTestAssemblies();
+            if (!sources || sources.length === 0) {
+                console.log('##vso[task.logissue type=warning;code=002004;]');
```
Comment (ShreyasRmsft) [inline Tasks/VsTestV2/nondistributedtest.ts]: can i change this to tl.warning? is it the same?

File: Tasks/VsTestV2/distributedtest.ts [hunk 2]
```diff
@@ -112,7 +100,18 @@ export class DistributedTest {
                 sourceFilter = ['**\\*', '!**\\obj\\*'];
             }
 
-            const sources = tl.findMatch(this.inputDataContract.TestSelectionSettings.SearchFolder, sourceFilter);
+            let sources;
+            try {
+                if (!utils.Helper.stringToBool(tl.getVariable('EnableVerboseFilesListing'))) {
```
Comment (nitin88) [inline Tasks/VsTestV2/distributedtest.ts]: why we need this? Enable/Disable -> there shouldn't be another secret flag to do.

File: Tasks/VsTestV2/nondistributedtest.ts [hunk 3]
```diff
@@ -116,30 +102,49 @@ export class NonDistributedTest {
...
-        return tl.findMatch(this.inputDataContract.TestSelectionSettings.SearchFolder, this.sourceFilter);
+
+        let sources;
+        try {
+            if (!utils.Helper.stringToBool(tl.getVariable('EnableVerboseFilesListing'))) {
+                tl.setVariable('System.Debug', 'false');
+            }
+            sources = tl.findMatch(this.inputDataContract.TestSelectionSettings.SearchFolder, this.sourceFilter);
+        } finally {
+            if (this.inputDataContract.Logging.DebugLogging) {
```
[the remark this answers is not in the document]
Comment (ShreyasRmsft) [reply Tasks/VsTestV2/nondistributedtest.ts]: can you please explain/elaborate?

File: Tasks/VsTestV2/nondistributedtest.ts [hunk 4]
```diff
@@ -116,30 +102,49 @@ export class NonDistributedTest {
...
         try {
             console.log(tl.loc('UserProvidedSourceFilter', this.sourceFilter.toString()));
-    
-            const sources = tl.findMatch(this.inputDataContract.TestSelectionSettings.SearchFolder, this.sourceFilter);
+
+            const sources = this.getTestAssemblies();
+            if (!sources || sources.length === 0) {
+                console.log('##vso[task.logissue type=warning;code=002004;]');
+                tl.warning(tl.loc('NoMatchingTestAssemblies', this.sourceFilter));
+                return;
```
[the remark this answers is not in the document]
Comment (ShreyasRmsft) [reply Tasks/VsTestV2/nondistributedtest.ts]: This refactoring is not yet complete. I still need to return null and add a check outside. Wanted a review on the turning on and off of the debug part

[the remark this answers is not in the document]
Comment (ShreyasRmsft) [reply Tasks/VsTestV2/nondistributedtest.ts]: Will pick this up in sometime when the customer query load is slightly lighter

**Comment by stephenmichaelf** [comment]
Closing due to age, please reopen if needed.

**Pull request closed without merging**